ccpp/data/GFS_typedefs.F90

```diff
@@ -4288,6 +4357,228 @@ subroutine control_initialize (Model, nlunit, fn_nml, me, master, &
       enddo
     endif
 
+    ! Tracer diagnostics indices and dimension size, which must be in
```

Comment:
I wonder whether we should put all the extended diagnostic logic into its own subroutine. Or leave it as-is and wait what EMC says.

Reply:
Seeing all the changes below, also routines `fill_dtidx` & Co, makes me wonder whether a completely separate module (file) to import in `GFS_typedefs` would be better.

Reply:
I'd rather not add new files until the upstream repository maintainers say they want more files.